### PR TITLE
move dialog settings to another script to fix exported game issues

### PR DESCRIPTION
**Pull request by skogach**
There's a problem when you try to export game and launch it, you'll get a grey screen. In the log there's errors about  `Unknown class: "EditorPlugin"` presumably because editor classes aren't exported. The problem is that `escoria-dialog-simple` plugin exports it's class name to other scripts which get settings via it. So I've moved settings to separate script that extends `Resource` same as `ESCProjectSettingsManager`. It fixes the problem.

### CODE REVIEW

**Comment by BHSDuncan** [comment]
Thanks for this. I have a couple comments, but first, `addons/escoria-dialog-simple/esc_dialo_simple_settings.gd` has a typo in the filename (I'm sure you can see it now).

The issue you bring up is valid but also highlights something I think we overlooked: The core has an explicit reference to a class located in a plugin, which is not a good idea, e.g. what if someone wants to implement their own dialog manager?

@StraToN @balloonpopper can you see any issues or challenges in moving `ESCDialogPlayer` (and its state machine) into the Simple Dialog addon? One thing we'll need to do is make sure that `ESCProjectSettingsManager` is somehow exposed so that the Simple Dialog addon can get at things like the speech folder, speech file extension, etc.

**Review by BHSDuncan — CHANGES_REQUESTED** [review]
`addons/escoria-dialog-simple/esc_dialo_simple_settings.gd` has a typo in the filename. Please correct.

File: addons/escoria-dialog-simple/esc_dialo_simple_settings.gd [hunk 1]
```diff
@@ -0,0 +1,20 @@
+extends Resource
+class_name ESCSimpleDialogSettingsManager
+
+const SETTINGS_ROOT="escoria/dialog_simple"
```
Comment (BHSDuncan) [inline addons/escoria-dialog-simple/esc_dialo_simple_settings.gd:4]:
```suggestion
const SETTINGS_ROOT = "escoria/dialog_simple"
```

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

**Comment by skogach** [comment]
I fixed typo and whitespaces, also I forgot to commit changes to project file, now it's there.

**Review by BHSDuncan — APPROVED** [review]

**Comment by StraToN** [comment]
> The issue you bring up is valid but also highlights something I think we overlooked: The core has an explicit reference to a class located in a plugin, which is not a good idea, e.g. what if someone wants to implement their own dialog manager?

We have two solutions: 
- either define a hook mechanism for any user to personalize some parts of the dialog manager. But then we'd need to define what hooks would be necessary for each needs.
- or allow a plugin to override the default behavior. Which would still require to design a clear API for this purpose.

> @StraToN @balloonpopper can you see any issues or challenges in moving ESCDialogPlayer (and its state machine) into the Simple Dialog addon? One thing we'll need to do is make sure that ESCProjectSettingsManager is somehow exposed so that the Simple Dialog addon can get at things like the speech folder, speech file extension, etc.

`ESCProjectSettingsManager` is a static class filled with static const values, so it can be used from anywhere, including a plugin.

I am a bit mixed about the idea of moving `ESCDialogPlayer` out of core. That would mean that any plugin that needs to use it will have to include it (as-is or edited for their own purpose). Again in that case I'd go in favor of some mechanism allowing one to override the default ESCDialogPlayer behavior if needed.

**Comment by BHSDuncan** [comment]
Let’s discuss further on Discord when you have time. On Feb 15, 2023, at 10:59 AM, Julian Murgia ***@***.***> wrote:﻿

The issue you bring up is valid but also highlights something I think we overlooked: The core has an explicit reference to a class located in a plugin, which is not a good idea, e.g. what if someone wants to implement their own dialog manager?

We have two solutions:

either define a hook mechanism for any user to personalize some parts of the dialog manager. But then we'd need to define what hooks would be necessary for each needs.
or allow a plugin to override the default behavior. Which would still require to design a clear API for this purpose.


@StraToN @balloonpopper can you see any issues or challenges in moving ESCDialogPlayer (and its state machine) into the Simple Dialog addon? One thing we'll need to do is make sure that ESCProjectSettingsManager is somehow exposed so that the Simple Dialog addon can get at things like the speech folder, speech file extension, etc.

ESCProjectSettingsManager is a static class filled with static const values, so it can be used from anywhere, including a plugin.
I am a bit mixed about the idea of moving ESCDialogPlayer out of core. That would mean that any plugin that needs to use it will have to include it (as-is or edited for their own purpose). Again in that case I'd go in favor of some mechanism allowing one to override the default ESCDialogPlayer behavior if needed.

**Review by StraToN — APPROVED** [review]

**Comment by StraToN** [comment]
@skogach I sincerely apologize for the time we took for reviewing and merging this PR. Thanks a lot!
Fixes https://github.com/godot-escoria/escoria-issues/issues/354
Closes https://github.com/godot-escoria/escoria-demo-game/pull/684

**Pull request merged**